src/aurora/textureatlasfile.cpp

```diff
@@ -0,0 +1,109 @@
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file
+ * A loader class for texture atlas xml files for dragonage and dragonage2
+ */
+
+#include "src/common/ustring.h"
+#include "src/common/readstream.h"
+#include "src/common/xml.h"
+#include "src/common/streamtokenizer.h"
+#include "src/common/strutil.h"
+
+#include "src/aurora/resman.h"
+#include "src/aurora/textureatlasfile.h"
+
+namespace Aurora {
+
+TextureAtlasFile::TextureAtlasFile(Common::UString name) {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource(name, kFileTypeXML));
+	load(*stream);
+}
+
+TextureAtlasFile::TextureAtlasFile(Common::SeekableReadStream &stream) {
+	load(stream);
+}
+
+bool TextureAtlasFile::isAtlasTexture(const Common::UString &texture) const {
+	return _atlasTextures.find(texture) != _atlasTextures.end();
+}
+
+void TextureAtlasFile::getAtlasTexture(const Common::UString &texture,
+                                       Common::UString &textureFile, float &x, float &y, float &w, float &h) const {
+	std::map<Common::UString, AtlasTexture>::const_iterator iter = _atlasTextures.find(texture);
+	if (iter == _atlasTextures.end()) {
+		throw Common::Exception("Atlas Texture %s not found", texture.c_str());
+	}
+
+	const AtlasTexture &atlasTexture = iter->second;
+	textureFile = atlasTexture.textureFile;
+	x = atlasTexture.x;
+	y = atlasTexture.y;
+	w = atlasTexture.w;
+	h = atlasTexture.h;
+}
+
+void TextureAtlasFile::load(Common::SeekableReadStream &stream) {
+	Common::XMLParser parser(stream);
+
+	const Common::XMLNode &node = parser.getRoot();
+
+	if(node.getName() != "AtlasData")
+		throw Common::Exception("Invalid tag, <AtlasData> expected, <%s> found", node.getName().c_str());
+
+	const Common::XMLNode::Children &atlasTextures = node.getChildren();
+	for (Common::XMLNode::Children::const_iterator c = atlasTextures.begin(); c != atlasTextures.end(); ++c) {
+		if ((*c)->getName() != "AtlasTexture")
+			throw Common::Exception("Invalid tag, <AtlasTexture> expected, <%s> found", (*c)->getName().c_str());
+
+		Common::UString atlasTexture = (*c)->getProperty("Name");
+
+		const Common::XMLNode::Children &sourceTextures = (*c)->getChildren();
+		for (Common::XMLNode::Children::const_iterator c = sourceTextures.begin();
+             c != sourceTextures.end(); ++c) {
```

Comment:
Here too. Use tabs to indent to the correct level of the for (i.e. two tabs here), then spaces to align the rest